src/Rules/Test.hs

```diff
@@ -63,13 +64,24 @@ testRules = do
         -- Execute the test target.
         buildWithCmdOptions env $ target (vanillaContext Stage2 compiler) RunTest [] []
 
+-- | Build extra programs required by testsuite
+needTestsuiteBuilders :: Action ()
+needTestsuiteBuilders = do
+    targets <- mapM (f Stage1) =<< testsuitePackages
+    need targets
+  where
+    f :: Stage -> Package -> Action FilePath
+    f stage pkg | isLibrary pkg = pkgConfFile (Context stage pkg (read "v"))
```

Comment:
I think you need `vanillaContext` here.

Reply:
Btw, are there testsuite-only libraries? If yes, are you sure `vanilla` is sufficient for the testsuite?

Reply:
This bit of code is taken directly from Rules.hs.
There are many errors coming from rts library (40). Although I still couldn't figure out from where errors are coming, I have added this part just in case.